### DISCUSSION TITLE
Support for Pimoroni Audio DAC SHIM

### DISCUSSION THREAD

**Discussion by ceejayemm** [rern/rAudio#190] [Fixed Issues] [answered]
@rern 

I have acquired this new DAC from Pimoroni for use with my RPi Zero W, see:

https://shop.pimoroni.com/products/audio-dac-shim-line-out

The details suggest that it should work with the HifiBerry DAC settings and I have configured rAudio-1 (2021020121) to use this DAC however with this selected the MPD page either cannot be displayed or appears to show no sound card installed. Could rAudio-1 be updated to use this DAC ?

Thanks 

Chris

**Comment by rern** [comment]
Try:
```sh
wget -O /srv/http/settings/system-i2s.json https://github.com/rern/rAudio-1/blob/UPDATE/srv/http/settings/system-i2s.json
wget -O /srv/http/bash/system.sh https://github.com/rern/rAudio-1/raw/UPDATE/srv/http/bash/system.sh
```

**Reply by ceejayemm** [reply]
Applied command lines as shown.  Now cannot open GPIO Devices / Audio- I2S options list to choose sound card device.

Chris

**Reply by rern** [reply]
Sorry. 1st link is html, not the script. Try this again.
```sh
wget -O /srv/http/settings/system-i2s.json https://github.com/rern/rAudio-1/raw/UPDATE/srv/http/settings/system-i2s.json
```

**Reply by ceejayemm** [reply]
@rern 
Thanks for the new command above which does now allow me to choose an I2S device on the System page. After enabling the new device here it asks me to reboot which I have done. After reboot the MPD page is available to me but the Output Device /  Mixer setting are missing. When I checked the /boot/config.txt file I found that the soundcard entry was incorrect:

dtoverlay=dtoverlay=hifiberry-dac

I corrected this to:

dtoverlay=hifiberry-dac

and rebooted. After reboot the MPD page gave me the following error:

Errors: Unexpected token , in JSON at position 307
{ "devices" : [{ "aplayname" : "hifiberry-dac" , "card" : 0 , "device" : 0 , "dop" : 0 , "hw" : "hw:0,0" , "hwmixer" : "( not available )" , "mixers" : 0 , "mixerdevices" : ["( not available )"] , "mixermanual" : false , "mixertype" : "none" , "name" : "hifiberry-dac" }] , "active" : true , "asoundcard" : ?, "audioaplayname" : "hifiberry-dac" , "audiooutput" : "hifiberry-dac" , "autoupdate" : false , "buffer" : false , "bufferval" : false , "bufferoutput" : false , "bufferoutputval" : false , "counts" : { "album": 314, "albumartist": 126, "artist": 560, "composer": 992, "date": 150, "genre": 33, "nas": 1, "sd": 0, "usb": 0, "song": 4789, "webradio": 21 } , "crossfade" : false , "crossfadeval" : false , "custom" : false , "ffmpeg" : true , "normalization" : false , "reboot" : "" , "replaygain" : false , "replaygainval" : "" , "soxr" : false , "soxrval" : "" , "version" : "0.22.4-2" }

It seems likely that this soundcard does not have a hardware mixer so I was hoping to use the MPD volume control instead. The sound card seems to installed and recognised by the RPi Zero:

rAudio:~  #  aplay -l
**** List of PLAYBACK Hardware Devices ****
card 0: sndrpihifiberry [snd_rpi_hifiberry_dac], device 0: HifiBerry DAC HiFi pcm5102a-hifi-0 [HifiBerry DAC HiFi pcm5102a-hifi-0]
  Subdevices: 1/1
  Subdevice #0: subdevice #0
rAudio:~  #

and 

rAudio:~  #  aplay -L
null
    Discard all samples (playback) or generate zero samples (capture)
default:CARD=sndrpihifiberry
    snd_rpi_hifiberry_dac, HifiBerry DAC HiFi pcm5102a-hifi-0
    Default Audio Device
sysdefault:CARD=sndrpihifiberry
    snd_rpi_hifiberry_dac, HifiBerry DAC HiFi pcm5102a-hifi-0
    Default Audio Device
rAudio:~  #

Thanks for your efforts

Chris

**Comment by rern** [comment]
Set I2S back to disabled and:
```sh
wget -O /srv/http/settings/system-i2s.json https://github.com/rern/rAudio-1/raw/UPDATE/srv/http/settings/system-i2s.json
wget -O /srv/http/bash/mpd-devices.sh https://github.com/rern/rAudio-1/raw/UPDATE/srv/http/bash/mpd-devices.sh
rm /etc/asound.conf
```

**Reply by ceejayemm** [reply]
@rern 
Thanks for these instructions. 

RAudio-1 now seems to recognise this Pimoroni DAC and I have configured the mixer in the MPD page but unfortunately still have no audio output. As a check I have also flashed a separate new card with the Pimoroni instructions for their Pirate Audio / Mopidy software and I also get no output from that installation either. I have contacted Pimoroni for a replacement so I will put this issue on hold until it arrives.

Thanks as always for you very timely support.

Chris

**Comment by ceejayemm** [comment, marked as answer]
@rern 
Replacement card arrived over the weekend and now works fine with the updates you made to rAudiio-1.

Many thanks

Chris
